src/list.js

```diff
@@ -171,6 +171,9 @@ Sk.builtin.list.prototype.tp$richcompare = function (w, op) {
             return true;
         }
 
+        if (Sk.__future__.python3) {
+            return Sk.builtin.NotImplemented.NotImplemented$;
```

Comment:
It is strange that we're returning these here? Instead of raising them?

Reply:
Yes, it is, but that's what the caller is expecting! I'm guessing we return a sentinel value because raising exceptions is expensive, and on this code path you can end up trying a lot of handlers before one of them works.

This is a `tp$` function -- ie it's in JS-land -- so Skulpt can make up whatever protocols it likes. (For example, `tp$getattr()` returns `undefined` rather than throwing an `AttributeError`, for similar reasons.)